cpp/include/raft/distance/detail/pairwise_matrix/kernel_sm60.cuh

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cstddef>
+#include <raft/core/operators.hpp>
+#include <raft/distance/detail/pairwise_distance_base.cuh>
+
+namespace raft::distance::detail {
+
+template <typename Policy,
+          bool row_major,
+          typename DataT,
+          typename AccT,
+          typename OutT,
+          typename IdxT,
+          typename opT,
+          typename FinOpT>
+__global__ __launch_bounds__(Policy::Nthreads, 2) void pairwise_matrix_kernel(const DataT* x,
+                                                                              const DataT* y,
+                                                                              const DataT* _xn,
+                                                                              const DataT* _yn,
+                                                                              IdxT m,
+                                                                              IdxT n,
+                                                                              IdxT k,
+                                                                              IdxT lda,
+                                                                              IdxT ldb,
+                                                                              IdxT ldd,
+                                                                              OutT* dOutput,
+                                                                              opT distance_op,
+                                                                              FinOpT fin_op)
+{
+  extern __shared__ char smem[];
+
+  // Wrap operator back into lambdas. This is temporary and should be removed. (TODO)
```

Comment:
This is done to avoid changing  `PairwiseDistances`?

Reply:
Rather than just putting a todo in the code, can we create a tracking issue for this and reference it here in the code please?

Reply:
I have added issue https://github.com/rapidsai/raft/issues/1323.